segmentation/utils/signal_extraction.py

```diff
@@ -1,36 +1,89 @@
 import numpy as np
+from skimage.measure import regionprops
 
 
-def simple_weighted_extraction(cell_coords, image_data):
+# TODO: work on weighted extraction and implement other discussed techniques of extraction
+def positive_pixels_extraction(cell_coords, image_data, threshold=0):
     """
-    Define a weighted extraction technique based on how close the coordinate is to the center
+    Extract channel counts by summing over the number of non-zero pixels in the cell,
+    improves on default_extraction by not distinguishing between pixel expression values
 
     Args:
-        cell_coords: tuples of values representing pixels within one cell
-        image_data: array containing channel counts
+        cell_coords (numpy): values representing pixels within one cell
+        image_data (xarray): array containing channel counts
 
     Returns:
-        np.array: sum of counts for each channel
+        channel_counts (numpy): sum of counts for each channel
     """
 
-    pass
+    # index indo image_data to get th channel values we're interested in
+    channel_values = image_data.values[tuple(cell_coords.T)]
+
+    # sum up based on a binary mask that is 1 if the expression value > threshold else 0
+    channel_counts = np.sum(channel_values > threshold, axis=0)
+
+    return channel_counts
+
+
+def center_weighting_extraction(cell_coords, image_data):
+    """
+    Extract channel counts by summing over weighted expression values based on distance from center,
+    improves upon default extraction by including a level of certainty/uncertainty
+
+    Args:
+        cell_coords (numpy): values representing pixels within one cell
+        image_data (xarray): array containing channel counts
+        segmentation_mask (numpy): array containing segmentation labels for each cell
+        cell_label (int): the cell number we wish to extract for each cell, important to distinguish
+            which centroid we want in regionprops because this function only extracts signal
+            for one cell, though we may choose to change this in the future...
+
+    Returns:
+        channel_counts (numpy): sum of counts for each channel
+    """
+
+    # we could use regionprops but it's probably better to use this method (ala A-Kag's)
+    # because it doesn't require you to know which cell_label to index into
+    centroid = np.sum(cell_coords, axis=0) / cell_coords.shape[0]
```

Comment:
Since we're already calculating regionprops for every cell in the image, we can just add centroid as one of the things to be computed and then pass the centroid coordinates into this function